__CFAllocatorRespectsHintZeroWhenAllocating not consistently declared extern C

when ForFoundationOnly.h and ForSwiftFoundationOnly.h end up being imported, compile fails due to different linkage of `__CFAllocatorRespectsHintZeroWhenAllocating()`.

suggest removing it from ForFoundationOnly.h or adding `CF_PRIVATE`.

```
<module-includes>:1:10: note: in file included from <module-includes>:1:
1 | #include "CoreFoundation.h"
  |          `- note: in file included from <module-includes>:1:
2 | 

/opt/swift-6.0.2-RELEASE-ubuntu24.04-aarch64/usr/lib/swift/CoreFoundation/CoreFoundation.h:77:10: note: in file included from /opt/swift-6.0.2-RELEASE-ubuntu24.04-aarch64/usr/lib/swift/CoreFoundation/CoreFoundation.h:77:
 75 | #include "CFConstantKeys.h"
 76 | 
 77 | #include "ForSwiftFoundationOnly.h"
    |          `- note: in file included from /opt/swift-6.0.2-RELEASE-ubuntu24.04-aarch64/usr/lib/swift/CoreFoundation/CoreFoundation.h:77:
 78 | 
 79 | #if TARGET_OS_OSX || TARGET_OS_IPHONE || TARGET_OS_WIN32 || TARGET_OS_LINUX

<unknown>:0: error: could not build C module 'CoreFoundation'
/opt/swift-6.0.2-RELEASE-ubuntu24.04-aarch64/usr/lib/swift/CoreFoundation/ForSwiftFoundationOnly.h:108:20: error: declaration of '__CFAllocatorRespectsHintZeroWhenAllocating' has a different language linkage
106 | _CF_EXPORT_SCOPE_BEGIN
107 | 
108 | CF_PRIVATE Boolean __CFAllocatorRespectsHintZeroWhenAllocating(CFAllocatorRef _Nullable allocator);
    |                    `- error: declaration of '__CFAllocatorRespectsHintZeroWhenAllocating' has a different language linkage
109 | 
110 | struct __CFSwiftObject {

/opt/swift-6.0.2-RELEASE-ubuntu24.04-aarch64/usr/lib/swift/CoreFoundation/ForFoundationOnly.h:71:9: note: previous declaration is here
 69 | #endif
 70 | 
 71 | Boolean __CFAllocatorRespectsHintZeroWhenAllocating(CFAllocatorRef _Nullable allocator);
    |         `- note: previous declaration is here
 72 | typedef CF_ENUM(CFOptionFlags, _CFAllocatorHint) {
 73 |     _CFAllocatorHintZeroWhenAllocating = 1
```

--- a/Sources/CoreFoundation/CFBase.c
+++ b/Sources/CoreFoundation/CFBase.c
@@ -790,7 +790,7 @@ void *__CFSafelyReallocateWithAllocator(CFAllocatorRef allocator, void *destinat
     return _CLANG_ANALYZER_IGNORE_NONNULL(reallocated);;
 }
 
-Boolean __CFAllocatorRespectsHintZeroWhenAllocating(CFAllocatorRef allocator) {
+CF_PRIVATE Boolean __CFAllocatorRespectsHintZeroWhenAllocating(CFAllocatorRef allocator) {
     return allocator == kCFAllocatorSystemDefault || allocator == kCFAllocatorMallocZone;
 }
 

--- a/Sources/CoreFoundation/include/ForFoundationOnly.h
+++ b/Sources/CoreFoundation/include/ForFoundationOnly.h
@@ -70,8 +70,6 @@ CF_EXPORT void *_Nonnull __CFSafelyReallocate(void * _Nullable destination, size
 CF_EXPORT void *_Nonnull __CFSafelyReallocateWithAllocator(CFAllocatorRef _Nullable, void * _Nullable destination, size_t newCapacity, CFOptionFlags options, void (^_Nullable reallocationFailureHandler)(void *_Nonnull original, bool *_Nonnull outRecovered));
 #endif
 
-Boolean __CFAllocatorRespectsHintZeroWhenAllocating(CFAllocatorRef _Nullable allocator);
-
 typedef CF_ENUM(CFOptionFlags, _CFAllocatorHint) {
     _CFAllocatorHintZeroWhenAllocating = 1
 };

--- a/Sources/CoreFoundation/include/ForSwiftFoundationOnly.h
+++ b/Sources/CoreFoundation/include/ForSwiftFoundationOnly.h
@@ -105,8 +105,6 @@
 
 _CF_EXPORT_SCOPE_BEGIN
 
-CF_PRIVATE Boolean __CFAllocatorRespectsHintZeroWhenAllocating(CFAllocatorRef _Nullable allocator);
-
 struct __CFSwiftObject {
     uintptr_t isa;
 };

--- a/Sources/CoreFoundation/internalInclude/CFInternal.h
+++ b/Sources/CoreFoundation/internalInclude/CFInternal.h
@@ -1057,6 +1057,8 @@ CF_INLINE CFAllocatorRef __CFGetAllocator(CFTypeRef cf) {	// !!! Use with CF typ
     return *(CFAllocatorRef *)((char *)cf - 16);
 }
 
+CF_PRIVATE Boolean __CFAllocatorRespectsHintZeroWhenAllocating(CFAllocatorRef allocator);
+
 /* !!! Avoid #importing objc.h; e.g. converting this to a .m file */
 struct __objcFastEnumerationStateEquivalent {
     unsigned long state;

